python/cudf/cudf/core/dataframe.py

```diff
@@ -6239,6 +6240,146 @@ def interleave_columns(self):
             {None: libcudf.reshape.interleave_columns([*self._columns])}
         )
 
+    @_cudf_nvtx_annotate
+    def eval(self, expr: "str", inplace: "bool" = False, **kwargs):
+        """Evaluate a string describing operations on DataFrame columns.
+
+        Operates on columns only, not specific rows or elements.  This allows
+        `eval` to run arbitrary code, which can make you vulnerable to code
+        injection if you pass user input to this function.
+
+        Parameters
+        ----------
+        expr : str
+            The expression string to evaluate.
+        inplace : bool, default False
+            If the expression contains an assignment, whether to perform the
+            operation inplace and mutate the existing DataFrame. Otherwise,
+            a new DataFrame is returned.
+        **kwargs
```

Comment:
Check the docstring rendering on `**kwargs` -- it may need something fancy like using a raw docstring `r"""` or similar to make Sphinx render this correctly.

Reply:
Good call, I always forget to switch to raw strings when I should. I wish there was a way to get a warning about asterisks the way that flake8 warns about backslashes, but since asterisks are valid rst it just thinks the docstring is fine...

Reply:
Is this something we need to worry about? Looking at the docs for [`mean`](https://docs.rapids.ai/api/cudf/nightly/api_docs/api/cudf.DataFrame.mean.html?highlight=mean#cudf.DataFrame.mean) for example, they are rendered correctly, even though we [don't](https://github.com/rapidsai/cudf/blob/branch-22.06/python/cudf/cudf/core/frame.py#L2812) use a raw string for the docstring AFAICT.

Reply:
Actually this looks fine. I suspect it's because numpydoc is handling this better than Napoleon, which is what you and I have used on a lot of previous projects.

Reply:
^ That's probably why this works. Very interesting. Sphinx has totally butchered this in several past projects. Could also be NumPy-style vs Google/Sphinx-style docstrings (not just the plugin, but the actual docstring format and how Sphinx handles it).

Reply:
No, I don't think it's just the style. We use NumPy-style docstrings on other projects where this issue definitely pops up. My guess is that the numpydoc package is doing a better job of special-casing *args and **kwargs in the context of numpy-style docstrings than Napoleon is.